Adds support for prettifying PromQL expression (#10544)
prometheus/prometheus: promql/parser/prettier_rules.md

# Prettifying PromQL expressions
This files contains rules for prettifying PromQL expressions.

Note: The current version of prettier does not preserve comments

### Keywords
`max_characters_per_line`: Maximum number of characters that will be allowed on a single line in a prettified PromQL expression

### Rules
1. A node exceeding the `max_characters_per_line` will qualify for split unless

>    1. It is a `SubqueryExpr` and `MatrixSelector`

Reviewer: Am I right?
```suggestion
   1. It is a `MatrixSelector`
```

Author: Yes, because a subquery contains 2 parts:
1. The internal expression
2. Time suffix

And if a subquery expression will exceed `max_characters_per_line`, then the internal expression will be split the least, meaning, from a user’s perspective, a subquery will qualify for split.

# Prettifying PromQL expressions
This files contains rules for prettifying PromQL expressions.

### Keywords

> `max_characters_per_line`: Maximum number of characters that will be allowed in a single stringified node

Reviewer: ```suggestion
`max_characters_per_line`: Maximum number of characters that will be allowed on a single line in a prettified PromQL expression
```

> ## Rules

Reviewer: Just to be consistent with the above

```suggestion
### Rules
```

1. A node exceeding the `max_characters_per_line` will qualify for split unless

>    1. It is a function call with one argument eg. `rate(foo[1m])`

Reviewer: What if the argument is big? Do we not split in that case?

Author: No, based on https://github.com/prometheus/prometheus/pull/10544#discussion_r880236631, we do not split if a call is with 1 argument.

Reviewer: I feel it should be broken into multiple lines. Because that one argument can be very big. For example here https://github.com/prometheus/prometheus/pull/10544#discussion_r880310693, it should be broken into multiple lines for easy reading

>    2. It is a `VectorSelector`, `SubqueryExpr` and `MatrixSelector`

Reviewer: Why do we not split for `SubqueryExpr`?

2. Nodes that are nested within another node will be prettified only if they exceed the `max_characters_per_line`

> 3. Following conditions do not involve splitting:

Reviewer: Can you club this with point 1? Both are talking about when you do not split, and VectorSelector is repeated.
